program/tests/duplicate_block_proof.rs

```diff
@@ -869,3 +958,176 @@ async fn double_report() {
+        .get_account(report_key)
+        .await
+        .unwrap()
+        .unwrap();
+    let violation_report_size = std::mem::size_of::<ViolationReport>();
+    let violation_report: &ViolationReport =
+        pod_from_bytes(&report_account.data[0..violation_report_size]).unwrap();
+
+    let expected_violation_report = ViolationReport {
+        version: ViolationReport::VERSION,
+        reporter,
+        destination,
+        epoch: PodU64::from(EPOCH),
+        pubkey: leader.pubkey(),
+        slot: PodU64::from(slot),
+        violation_type: ProofType::DuplicateBlockProof.into(),
+        proof_account: account.pubkey(),
+    };
+    assert_eq!(*violation_report, expected_violation_report);
+
+    // Verify that the proof was also serialized to the account
+    let proof =
+        DuplicateBlockProofData::unpack_proof(&report_account.data[violation_report_size..])
+            .unwrap();
+    assert_eq!(duplicate_proof, proof);
+
+    // Close the report should fail as only 0 epochs have passed
+    let err = close_report(
+        &mut context,
+        report_key,
+        destination,
+        report_account.lamports,
+        0,
+    )
+    .await
+    .unwrap_err()
+    .unwrap();
+    let TransactionError::InstructionError(0, InstructionError::Custom(code)) = err else {
+        panic!("Invalid error {err:?}");
+    };
+    let err: SlashingError = SlashingError::decode_custom_error_to_enum(code).unwrap();
+    assert_eq!(err, SlashingError::CloseViolationReportTooSoon);
+
+    // Close the report should fail as only 1 epochs have passed
+    setup_clock(&mut context).await;
+    let err = close_report(
+        &mut context,
+        report_key,
+        destination,
+        report_account.lamports,
+        1,
+    )
+    .await
+    .unwrap_err()
+    .unwrap();
+    let TransactionError::InstructionError(0, InstructionError::Custom(code)) = err else {
+        panic!("Invalid error {err:?}");
+    };
+    let err: SlashingError = SlashingError::decode_custom_error_to_enum(code).unwrap();
+    assert_eq!(err, SlashingError::CloseViolationReportTooSoon);
+
+    // Close the report should fail as only 2 epochs have passed
+    setup_clock(&mut context).await;
+    let err = close_report(
+        &mut context,
+        report_key,
+        destination,
+        report_account.lamports,
+        2,
+    )
+    .await
+    .unwrap_err()
+    .unwrap();
+    let TransactionError::InstructionError(0, InstructionError::Custom(code)) = err else {
+        panic!("Invalid error {err:?}");
+    };
+    let err: SlashingError = SlashingError::decode_custom_error_to_enum(code).unwrap();
+    assert_eq!(err, SlashingError::CloseViolationReportTooSoon);
+
+    // Close report should fail with invalid destination account
+    let err = close_report(
+        &mut context,
+        report_key,
+        Pubkey::new_unique(),
+        report_account.lamports,
+        3,
+    )
+    .await
+    .unwrap_err()
+    .unwrap();
```

Comment:
Not a big deal, but this is missing a call to `setup_clock` before calling `close_report`